### ISSUE TITLE
Fix handling of empty sets in set matching metrics

### ISSUE THREAD

**Issue by wgantt** [wanmok/metametric#23]
Hi @wanmok and @ctongfei. In using this package for another project, I noticed that the current implementation of the set matching metrics gives undesired behavior in the case where:
- matching is one-to-one
- either the predicted or the reference set (but not both) is empty
- the inner metric is _not_ discrete (so [this line](https://github.com/wanmok/metametric/blob/main/src/metametric/core/matching.py#L25) is not triggered)

In such a case, it’s possible to reach [this line](https://github.com/wanmok/metametric/blob/main/src/metametric/core/matching.py#L29) with a gram “matrix” that is a 1D array rather than a 2D array, yielding an error like the following when the call to `spo.linear_sum_assignment` is made:

```
File "/brtx/601-nvme2/wgantt/.conda/envs/ote/lib/python3.10/site-packages/metametric/core/matching.py", line 30, in score
    row_idx, col_idx = spo.linear_sum_assignment(
ValueError: expected a matrix (2-D array), got a 1 array
```

It feels like we should just add logic to return 0 if we reach [here](https://github.com/wanmok/metametric/blob/main/src/metametric/core/matching.py#L28) and either `x` or `y` is empty. What do you think?

**Comment by ctongfei** [comment]
@wgantt I think that you are right. If either `x` or `y` is empty, we should bypass the following logic and just return `0`.

**Comment by ctongfei** [comment]
Fixed by #24 .
